api/src/main/java/com/github/eyefloaters/console/api/model/Topic.java

```diff
@@ -114,38 +115,81 @@ public ListResponse(List<Topic> data, ListRequestContext<Topic> listSupport) {
     }
 
     @Schema(name = "TopicResponse")
-    public static final class SingleResponse extends DataResponse<TopicResource> {
+    public static final class SingleResponse extends DataSingleton<Topic> {
         public SingleResponse(Topic data) {
-            super(new TopicResource(data));
+            super(data);
         }
     }
 
-    @Schema(name = "Topic")
-    public static final class TopicResource extends Resource<Topic> {
-        public TopicResource(Topic data) {
-            super(data.id, "topics", data);
+    @JsonFilter("fieldFilter")
+    static class Attributes {
+        @JsonProperty
+        String name;
+
+        @JsonProperty
+        boolean internal;
+
+        @JsonProperty
+        @Schema(implementation = Object.class, oneOf = { PartitionInfo[].class, Error.class })
+        Either<List<PartitionInfo>, Error> partitions;
+
+        @JsonProperty
+        @Schema(implementation = Object.class, oneOf = { String[].class, Error.class })
+        Either<List<String>, Error> authorizedOperations;
+
+        @JsonProperty
+        @Schema(implementation = Object.class, oneOf = { ConfigEntry.ConfigEntryMap.class, Error.class })
+        Either<Map<String, ConfigEntry>, Error> configs;
+
+        Attributes(String name, boolean internal) {
+            this.name = name;
+            this.internal = internal;
         }
-    }
 
-    String name;
-    boolean internal;
-    @JsonIgnore
-    String id;
+        /**
+         * Calculates the record count for the entire topic as the sum
+         * of the record counts of each individual partition. When the partitions
+         * are not available, the record count is null.
+         *
+         * @return the sum of the record counts for all partitions
+         */
+        public Long getRecordCount() {
+            return partitions.getOptionalPrimary()
+                .map(Collection::stream)
+                .map(p -> p.map(PartitionInfo::getRecordCount)
+                        .filter(Objects::nonNull)
+                        .reduce(0L, Long::sum))
+                .orElse(null);
+        }
 
-    @Schema(implementation = Object.class, oneOf = { PartitionInfo[].class, Error.class })
-    Either<List<PartitionInfo>, Error> partitions;
+        @Schema(readOnly = true, description = """
+                The total size, in bytes, of all log segments local to the leaders
+                for each of this topic's partition replicas.
+                Or null if this information is not available.
 
-    @Schema(implementation = Object.class, oneOf = { String[].class, Error.class })
-    Either<List<String>, Error> authorizedOperations;
+                When support for tiered storage (KIP-405) is available, this property
+                may also include the size of remote replica storage.
+                """)
+        public BigInteger getTotalLeaderLogBytes() {
```

Comment:
Do you want to stick with `getFoo()` method naming here, given that elsewhere you've been using `foo()`-style naming? Some of the call sites look a little inconsistent with `foo.bar().getBaz().quux()`.

Reply:
I do want to switch entirely to the non-bean style (#43), but it just hasn't happened yet. I'll be sure to get to it in the next few weeks in a purely non-functional change.